org.mixedrealitytoolkit.uxcore/Button/PressableButton.cs

```diff
@@ -148,6 +152,28 @@ public SpaceMode DistanceSpaceMode
         [Tooltip("Speed for extending the moving button visuals when selected by a non-touch source.")]
         private float extendSpeed = 0.5f;
 
+        [field: SerializeField, Tooltip("Enables (true) or disables (false) the Canvas Rounded Rect on proximity only.")]
+        public bool EnableOnProximityOnlyCanvasRoundedRect { get; set; } = false;
+
+        [field: SerializeField, Tooltip("Enables (true) or disables (false) the Front plate Raw Image on proximity only.")]
+        public bool EnableOnProximityOnlyFrontPlateRawImage { get; set; } = false;
+
+        /// <summary>
+        /// ActionButton's Front plate.
+        /// </summary>
+        [SerializeField]
+        [Tooltip("ActionButton's Front plate.")]
+        private GameObject frontPlate;
+
+        /// <summary>
+        /// ActionButton's Front plate.
+        /// </summary>
+        public GameObject FrontPlate
```

Comment:
I thought we were going down the path of using events to show or hide elements?    Why have you decided to go down this path?

I sort of like using events, so the button logic doesn't need to be aware of the visuals of button.   See `remarks`....a core principle of this component is to make no assumptions about visuals.  If we start making a strong connection to visuals in this class, we risk breaking the contract we had with developers.

Reply:
In the last commit I switched to the new behavior is to enable/disable by proximity any Component in an array populated in Editor as per new requirements.  I think the need to hook to event handlers won't be needed anymore.  Resolving this conversation but please feel free to reopen if considered necessary.